api/provider/aws/environment_scaler.go

```diff
@@ -1,14 +1,758 @@
+				end = len(serviceARNs)
+			}
+
+			describeServicesInput := &ecs.DescribeServicesInput{}
+			describeServicesInput.SetCluster(clusterName)
+			describeServicesInput.SetServices(serviceARNs[i:end])
+
+			output, err := e.Client.ECS.DescribeServices(describeServicesInput)
+			if err != nil {
+				return nil, err
+			}
+
+			services = append(services, output.Services...)
+		}
+	}
+
+	for _, service := range services {
+		deployIDCopies := map[string]int64{}
+		for _, d := range service.Deployments {
+			desiredCount := aws.Int64Value(d.DesiredCount)
+			runningCount := aws.Int64Value(d.RunningCount)
+			pendingCount := aws.Int64Value(d.PendingCount)
+			if numPending := desiredCount - (runningCount + pendingCount); numPending > 0 {
+				deployIDCopies[aws.StringValue(d.Id)] = numPending
+			}
+		}
+
+		if len(deployIDCopies) == 0 {
+			continue
+		}
+
+		// iterate through deploys
+		for deployID := range deployIDCopies {
+			c, err := e.getContainerResourceFromDeploy(deployID)
+			if err != nil {
+				return nil, err
+			}
+
+			resourceConsumers = append(resourceConsumers, c...)
+		}
+	}
+
+	return resourceConsumers, nil
+}
+
+func (e *EnvironmentScaler) getResourceConsumers_TasksInECS(clusterName string) ([]ResourceConsumer, error) {
+	var (
+		taskARNs          []string
+		resourceConsumers []ResourceConsumer
+	)
+
+	fn := func(output *ecs.ListTasksOutput, lastPage bool) bool {
+
+		for _, taskARN := range output.TaskArns {
+			taskARNs = append(taskARNs, aws.StringValue(taskARN))
+		}
+
+		return !lastPage
+	}
+
+	startedBy := e.Config.Instance()
+	for _, status := range []string{ecs.DesiredStatusRunning, ecs.DesiredStatusStopped} {
+		input := &ecs.ListTasksInput{}
+		input.SetCluster(clusterName)
+		input.SetDesiredStatus(status)
+		input.SetStartedBy(startedBy)
+		if err := e.Client.ECS.ListTasksPages(input, fn); err != nil {
+			return nil, err
+		}
+	}
+
+	for _, taskARN := range taskARNs {
+		task, err := e.TaskProvider.Read(taskARN)
+		if err != nil {
+			return nil, err
+		}
+
+		taskResourceConsumers, err := e.getContainerResourceFromDeploy(task.DeployID)
+		if err != nil {
+			return nil, err
+		}
+
+		resourceConsumers = append(resourceConsumers, taskResourceConsumers...)
+	}
+
+	return resourceConsumers, nil
+}
+
+func (e *EnvironmentScaler) getResourceConsumers_TasksInJobs(clusterName string) ([]ResourceConsumer, error) {
+	var resourceConsumers []ResourceConsumer
+	jobs, err := e.JobStore.SelectAll()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, job := range jobs {
+		if job.Type == models.CreateTaskJob {
+			// TODO: maybe remove pending check here
+			if job.Status == models.PendingJobStatus || job.Status == models.InProgressJobStatus {
+				var req models.CreateTaskRequest
+				if err := json.Unmarshal([]byte(job.Request), &req); err != nil {
+					return nil, err
+				}
+
+				if req.EnvironmentID == clusterName {
+					// note that this isn't exact if the job has started some, but not all of the tasks
```

Comment:
that comment no longer applies because tasks don't have copies

Reply:
✅